cairo/src/context.rs

```diff
@@ -343,7 +342,7 @@ impl Context {
         unsafe { ffi::cairo_clip_preserve(self.0.as_ptr()) }
     }
 
-    pub fn clip_extents(&self) -> (f64, f64, f64, f64) {
+    pub fn clip_extents(&self) -> Result<(f64, f64, f64, f64), Error> {
```

Comment:
Why can this and `in_clip` fail? They're just querying or not?

Reply:
Because if the the context is in an error state, the output wouldn't make much sense

Reply:
Hm ok!